src/helpers/graphHelper.js

```diff
@@ -21,10 +21,15 @@ exports.createUser = function (
         onPremisesSamAccountName: user,
         passwordProfile: { password }
     })
-    userSettings.addUserToCreatedUsersList(user, password, displayName, email)
+
     return httpHelper
         .postGraph('users', 'admin', body)
-        .then(res => httpHelper.checkStatus(res, 'Failed while creating user'))
+        .then((res) => {
+            if (res.ok) {
+                userSettings.addUserToCreatedUsersList(user, password, displayName, email)
+            }
+            httpHelper.checkStatus(res, 'Failed while creating user')
```

Comment:
maybe adding to created list after this check if better. status code also should be good before push

Reply:
@kiranparajuli589 in my view order doesn't matter because we res.ok check the response and only add user in array